source/kernel/src/trace.rs

```diff
@@ -116,17 +220,28 @@ impl Collect for SerialCollector {
             meta: metadata.as_serde(),
         });
 
-        if sent {
-            tracing_core_02::Interest::always()
-        } else {
-            // if we couldn't send the metadata, skip this callsite, because the
-            // consumer will not be able to understand it without its metadata.
-            tracing_core_02::Interest::never()
+        // If we couldn't send the metadata, skip this callsite, because the
+        // consumer will not be able to understand it without its metadata.
+        if !sent {
+            return tracing_core_02::Interest::never();
+        }
+
+        // Due to the fact that the collector uses `bbq` internally, we must
```

Comment:
This *feels* a little icky, because this is a platform-specific impl detail - melpo doesn't use bbqueue for the tracing collection.

Might be worth instead to::

* Provide bbqueue methods that don't emit traces (and use them in lichee-rv)
* Start an "integration guide" that tells people to use something that doesn't emit traces when implementing tracing

I don't think this is a today blocker, but if we merge this as-is it might be good to throw up an issue to fix this someday.

Reply:
Well, *this* collector does use `bbqueue` --- so tif we are ever using the collector in this file, we are using bbqueue.